File: Cpp/11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size()-1;
        int ans = 0;
        while(i < j){
            ans = max(ans,min(height[i], height[j]) * abs(j-i));
            if(height[i] > height[j]){
                j--;
            }else if(height[i] < height[j]){
                i++;
            }else{
                i++;
                j--;
            }
        }
        return ans;
        
    }
};

File: Python/11.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)-1
        sol = 0
        while l < r:
            water = (min(nums[l] , nums[r]) * (r-l))
            if nums[l] < nums[r]:
                l+=1
            else:
                r-=1
            sol = max(sol, water)
        return sol
